src/AppInstallerCLICore/Workflows/ManifestComparator.cpp

```diff
@@ -295,26 +315,38 @@ namespace AppInstaller::CLI::Workflow
                 }
             }
 
-            std::string ExplainInapplicable(const Manifest::ManifestInstaller& installer) override
-            {
-                std::string result = "InstallerType does not match required type: ";
-                result += InstallerTypeToString(installer.EffectiveInstallerType());
-                result += "Required InstallerTypes: ";
-                result += m_requirementAsString;
-                return result;
-            }
-
             bool IsFirstBetter(const Manifest::ManifestInstaller& first, const Manifest::ManifestInstaller& second) override
             {
-                // TODO: Current implementation assumes there is only a single installer type requirement. This needs to be updated
-                // once multiple installerType requirements and preferences are accepted.
-                UNREFERENCED_PARAMETER(first);
-                UNREFERENCED_PARAMETER(second);
-                return true;
+                if (m_preference.empty())
+                {
+                    return false;
+                }
+
+                bool isFirstInstallerTypePreferred =
+                    ContainsInstallerType(m_preference, first.EffectiveInstallerType()) ||
+                    ContainsInstallerType(m_preference, first.BaseInstallerType);
+
+                bool isSecondInstallerTypePreferred = 
+                    ContainsInstallerType(m_preference, second.EffectiveInstallerType()) ||
+                    ContainsInstallerType(m_preference, second.BaseInstallerType);
+
+                if (isFirstInstallerTypePreferred && isSecondInstallerTypePreferred)
+                {
+                    return false;
+                }
+
+                if (isFirstInstallerTypePreferred != isSecondInstallerTypePreferred)
+                {
+                    return isFirstInstallerTypePreferred;
+                }
+
+                return false;

```

Comment:
```suggestion
               return isFirstInstallerTypePreferred > isSecondInstallerTypePreferred
```
`(true > false)` returns `true` - equivalent to `isFirstInstallerTypePreferred != isSecondInstallerTypePreferred`
`(true > true)` returns `false` - equivalent to `isFirstInstallerTypePreferred && isSecondInstallerTypePreferred`
`(false > false)` returns `false` - equivalent to when neither if statement is matched
`(false > true)` returns `false` - equivalent to `isFirstInstallerTypePreferred != isSecondInstallerTypePreferred`

Reply:
Went with Flor's suggestions as I don't want to hide too much behind a boolean comparison as it would make it hard to reason.